sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -233,6 +233,18 @@ private[hive] case class HiveGenericUDF(funcWrapper: HiveFunctionWrapper, childr
     udfType != null && udfType.deterministic()
   }
 
+  override def semanticEquals(other: Expression): Boolean = {
```

Comment:
Can you comment why the version in Expression doesn't work? What about it is too strict?